janus_core/processing/observables.py

```diff
@@ -2,73 +2,254 @@
 
 from __future__ import annotations
 
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
 from ase import Atoms, units
 
+if TYPE_CHECKING:
+    from janus_core.helpers.janus_types import SliceLike
+
+from janus_core.helpers.utils import slicelike_to_startstopstep
+
+
+# pylint: disable=too-few-public-methods
+class Observable(ABC):
+    """
+    Observable data that may be correlated.
+
+    Parameters
+    ----------
+    atoms_slice : list[int] | SliceLike | None = None
+        A slice of atoms to observe.
+    """
+
+    def __init__(self, atoms_slice: list[int] | SliceLike | None = None):
+        """
+        Initialise an observable with a given dimensionality.
+
+        Parameters
+        ----------
+        atoms_slice : list[int] | SliceLike | None = None
+            A slice of atoms to observe.
```

Comment:
```suggestion
        atoms_slice : list[int] | SliceLike | None
            A slice of atoms to observe. By default all atoms are included.
```
Or similar i.e. describe what happens if `None`